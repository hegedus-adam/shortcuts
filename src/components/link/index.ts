export { renderLink } from './link.component';
