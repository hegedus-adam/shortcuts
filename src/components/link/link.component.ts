interface ILink {
  id: number
  label: string
  url: string
}

export function renderLink(link: ILink) {
  const li = document.createElement('li');

  const anchor = document.createElement('a');
  anchor.href = link.url;
  anchor.innerHTML = link.label;
  anchor.className = 'link';

  li.appendChild(anchor);

  return li;
}
