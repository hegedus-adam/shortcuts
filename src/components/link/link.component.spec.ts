import { renderLink } from './link.component';
import { JSDOM } from 'jsdom';

const dom = new JSDOM();
global.document = dom.window.document;
global.window = dom.window;

describe('renderLink', () => {
  it('should append link', () => {
    const link = {
      id: 0,
      label: 'Test Label',
      url: 'https://somewhere.to',
    };

    const actual = renderLink(link);

    expect(actual).toMatchSnapshot();
  });
});
