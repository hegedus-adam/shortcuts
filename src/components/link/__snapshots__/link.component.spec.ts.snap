// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderLink should append link 1`] = `
<li>
  <a
    class="link"
    href="https://somewhere.to"
  >
    Test Label
  </a>
</li>
`;
